runtime/onert/backend/trix/ops/BulkLayer.cc

```diff
@@ -84,28 +117,48 @@ void BulkLayer::run()
   _dev_context->setDataInfo<const IPortableTensor>(&in_info, _inputs);
   _dev_context->setDataInfo<IPortableTensor>(&out_info, _outputs);
 
-  input_buffers input_buf;
-  output_buffers output_buf;
-  _dev_context->setBuffer<const IPortableTensor>(&input_buf, _inputs);
-  _dev_context->setBuffer<IPortableTensor>(&output_buf, _outputs);
+  input_buffers input_buf[2];
+  output_buffers output_buf[2];
 
-  if (setNPU_requestData(_dev_context->getDev(), req_id, &input_buf, &in_info, &output_buf,
-                         &out_info))
+  if (is_batch_execution)
   {
-    throw std::runtime_error("Unable to create NPU request for model id (" +
-                             std::to_string(_model_id) + ")");
+    // TODO: Support for general number of cores(>2)
+    // Here we assume that 2 trix cores
+    for (int i = 0; i < (batch_size); i = i + 2)
+    {
+      _dev_context->setBuffer<const IPortableTensor>(&input_buf[0], _inputs, batch_size, i);
+      _dev_context->setBuffer<IPortableTensor>(&output_buf[0], _outputs, batch_size, i);
+
+      _dev_context->setBuffer<const IPortableTensor>(&input_buf[1], _inputs, batch_size, i + 1);
+      _dev_context->setBuffer<IPortableTensor>(&output_buf[1], _outputs, batch_size, i + 1);
+
+      auto f0 = std::async(std::launch::async, &single_job, _dev_context->getDev(0), req_id[0],
+                           &input_buf[0], &in_info, &output_buf[0], &out_info);
+      if (i + 1 < batch_size) // ignore last job if batch_size is odd number
+      {
+        auto f1 = std::async(std::launch::async, &single_job, _dev_context->getDev(1), req_id[1],
+                             &input_buf[1], &in_info, &output_buf[1], &out_info);
+        f1.wait();
+      }
+      f0.wait();
+    }
```

Comment:
(Q) Does the `std::async` function replace what `single_job` function does?

Reply:
I don't 100% understand the meaning of _replace_. The main thread uses `std::async` to run `single_job` asynchronously. Internally, `std::async` will create worker thread and run `single_job` though its working depends on its implementation.